presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowPageUtils.java

```diff
@@ -0,0 +1,929 @@
+    {
+        if (value instanceof Boolean) {
+            type.writeBoolean(builder, (Boolean) value);
+        }
+        else {
+            throw new IllegalArgumentException("Unsupported type for BooleanType: " + value.getClass());
+        }
+    }
+
+    public static void handleDecimalType(DecimalType type, BlockBuilder builder, Object value)
+    {
+        if (value instanceof BigDecimal) {
+            BigDecimal decimalValue = (BigDecimal) value;
+            if (type.isShort()) {
+                // Handle ShortDecimalType
+                long unscaledValue = decimalValue.unscaledValue().longValue();
+                type.writeLong(builder, unscaledValue);
+            }
+            else {
+                // Handle LongDecimalType
+                Slice slice = Decimals.encodeScaledValue(decimalValue);
+                type.writeSlice(builder, slice);
+            }
+        }
+        else if (value instanceof Long) {
+            // Direct handling for ShortDecimalType using long
+            if (type.isShort()) {
+                type.writeLong(builder, (Long) value);
+            }
+            else {
+                throw new IllegalArgumentException("Long value is not supported for LongDecimalType.");
+            }
+        }
+        else {
+            throw new IllegalArgumentException("Unsupported type for DecimalType: " + value.getClass());
+        }
+    }
+
+    public static void handleArrayType(ArrayType type, BlockBuilder builder, Object value)
+    {
+        Type elementType = type.getElementType();
+        BlockBuilder arrayBuilder = builder.beginBlockEntry();
+        for (Object element : (Iterable<?>) value) {
+            appendValueToBuilder(elementType, arrayBuilder, element);
+        }
+        builder.closeEntry();
+    }
+
+    public static void handleRowType(RowType type, BlockBuilder builder, Object value)
+    {
+        List<Object> rowValues = (List<Object>) value;
+        BlockBuilder rowBuilder = builder.beginBlockEntry();
+        List<RowType.Field> fields = type.getFields();
+        for (int i = 0; i < fields.size(); i++) {
+            Type fieldType = fields.get(i).getType();
+            appendValueToBuilder(fieldType, rowBuilder, rowValues.get(i));
+        }
+        builder.closeEntry();
+    }
+
+    public static void handleDateType(Type type, BlockBuilder builder, Object value)
+    {
+        if (value instanceof java.sql.Date || value instanceof java.time.LocalDate) {
+            int daysSinceEpoch = (int) (value instanceof java.sql.Date
+                    ? ((java.sql.Date) value).toLocalDate().toEpochDay()
+                    : ((java.time.LocalDate) value).toEpochDay());
+            type.writeLong(builder, daysSinceEpoch);
+        }
+        else {
+            throw new IllegalArgumentException("Unsupported type for DateType: " + value.getClass());
+        }
+    }
+
+    public static void handleTimestampType(Type type, BlockBuilder builder, Object value)
```

Comment:
instead of "handle...." name, wouldn't "writeTimestamp...." name be apt? Similarly for other "handle...." methods.

Reply:
done